taichi/backends/cuda/cuda_driver.cpp

```diff
@@ -78,4 +61,48 @@ CUDADriver &CUDADriver::get_instance() {
   return get_instance_without_context();
 }
 
+CUDADriverBase::CUDADriverBase() {
+  disabled_by_env_ = (get_environ_config("TI_ENABLE_CUDA", 1) == 0);
+  if (disabled_by_env_) {
+    TI_TRACE("CUDA driver disabled by enviroment variable \"TI_ENABLE_CUDA\".");
+    return;
+  }
+}
+
+void CUDADriverBase::load_lib(std::string lib_linux, std::string lib_windows) {
+#if defined(TI_PLATFORM_LINUX)
+  auto lib_name = lib_linux;
+#elif defined(TI_PLATFORM_WINDOWS)
+  auto lib_name = lib_windows;
+#else
+  static_assert(false, "Taichi CUDA driver supports only Windows and Linux.");
+#endif
+
+  loader_ = std::make_unique<DynamicLoader>(lib_name);
+  if (!loader_->loaded()) {
+    TI_WARN("{} lib not found.", lib_name);
+    return;
+  } else {
+    TI_TRACE("{} loaded!", lib_name);
+  }
+}
+
+CUSPARSEDriver::CUSPARSEDriver() {
```

Comment:
nit: I'd suggest not to do this in the constructor, becuase if the lib fails to load, we get a broken/zombie object. A safer approach could be to load DLL in a function. If that fails, returns the exception/error to Python. Otherwise move that loaded lib into `CuSolverDriver`.

Also, we need to make sure not accidentally invoke `CUSPARSEDRiver::get_instance()` when we don't need the sparse matrix, otherwise Taichi could be broken if cuSparse is not installed.

I know this is not a newly introduced problem, so this is just something to think about :-)